### PR TITLE
program: add check_order_constraints

### PR DESCRIPTION
**Pull request by crispheaney**
the idea is to check that current position + open bids < max position and current position + open asks is > min position

### CODE REVIEW

File: ts/sdk/src/jitProxyClient.ts [hunk 1]
```diff
@@ -135,4 +143,41 @@ export class JitProxyClient {
...
+		subAccountId =
+			subAccountId !== undefined
+				? subAccountId
+				: this.driftClient.activeSubAccountId;
+
+		const readablePerpMarketIndex = [];
+		const readableSpotMarketIndexes = [];
+		for (const orderConstraint of orderConstraints) {
+			if (isVariant(orderConstraint.marketType, 'perp')) {
+				readablePerpMarketIndex.push(orderConstraint.marketIndex);
```
Comment (0xbigz) [inline ts/sdk/src/jitProxyClient.ts:163]: should filter for duplicates?

File: programs/jit-proxy/src/lib.rs [hunk 2]
```diff
@@ -237,6 +299,73 @@ pub enum PriceType {
...
+
+        Ok(())
+    }
+}
+
+#[derive(Clone, Copy, BorshSerialize, BorshDeserialize, PartialEq, Debug, Eq)]
+pub enum MarketType {
+    Perp,
+    Spot,
+}
```
Comment (0xbigz) [inline programs/jit-proxy/src/lib.rs:358]: why did you have to do recreate this rather than import?

Comment (crispheaney) [reply programs/jit-proxy/src/lib.rs:358]: annoying anchor thing, to get MarketType in jit idl, needs to be declared here

File: ts/sdk/src/jitProxyClient.ts (hunk 1 above)
Comment (crispheaney) [reply ts/sdk/src/jitProxyClient.ts:163]: `getRemainingAccounts` will handle duplicate market indexes

**Pull request merged**